File: source/_code_examples/websocket_connection.js
// In Javascript, you can use the built-in WebSocket API like shown below
// https://developer.mozilla.org/en-US/docs/Web/API/WebSocket

const socket = new WebSocket("https://my-kindafax-instance.example.com");

socket.addEventListener("open", event => {

    // When the connection opens. Send a message back to the server.
    socket.send("Hello, World!");
})

socket.addEventListener("message", event => {
    
    // Each time the client receives a message, log it to the console
    console.log(event.data);
})

socket.addEventListener("close", event => {

    // Log a message when the connection closes
    console.log("WebSocket Connection Closed");
})

socket.addEventListener("error", error => {

    // Log when the connection errors.
    console.error("Error with socket", error);
})

File: source/_code_examples/websocket_connection.py
import asyncio      # builtin
import websockets   # pip install websockets

WEBSOCKET_URL = "https://my-kindafax-instance.example.com"

async def main():

    # Open a connection, with a context handler.
    async with websockets.connect(WEBSOCKET_URL) as websocket:

        # Send a message to the server.
        await websocket.send("Hello World!")

        # Wait for a response back.
        response = await websocket.recv()
        print(response) # Print the response to the console


# Run asynchronously using asyncio
asyncio.run(main())